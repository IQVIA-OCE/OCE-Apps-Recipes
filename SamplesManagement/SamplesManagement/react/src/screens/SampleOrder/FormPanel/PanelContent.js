import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select, Autocomplete } from 'apollo-react-native';
import { useFormikContext } from 'formik';
import { useFetcher, useHandleData } from '../../../hooks';
import { fetchUserLocations } from '../../../api/SampleOrder';
import {
  normalizeUsers,
  normalizeLocations,
  getFieldError,
  getFieldHelperText,
} from '../utils';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, handleChange, setFieldValue, errors, touched } = context;

  const [locations] = useFetcher(async () => {
    return await fetchUserLocations(values.fields.user.Id);
  }, normalizeLocations);

  return (
    <View style={styles.container}>
      <View style={styles.col}>
        {useHandleData(locations)(data => {
          return readonly ? (
            <TextInput
              label="Ship To"
              value={values.fields.shipTo ? values.fields.shipTo.label : ''}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Select
              label="Ship To"
              placeholder={'-None-'}
              options={data}
              value={values.fields.shipTo}
              onChange={val => setFieldValue('fields.shipTo', val)}
              fullWidth
              style={{ width: '100%', marginBottom: 15 }}
              error={getFieldError('shipTo', errors, touched)}
              helperText={getFieldHelperText('shipTo', errors, touched)}
              required
            />
          );
        })}
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyWithBorder : styles.field}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  readonlyField: {
    marginBottom: 18,
  },
  readonlyWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
