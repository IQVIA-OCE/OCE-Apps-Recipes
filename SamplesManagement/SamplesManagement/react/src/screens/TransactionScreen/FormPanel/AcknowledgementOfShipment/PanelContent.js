import React from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select } from '@oce-apps/apollo-react-native';
import { useFormikContext } from 'formik';
import DateField from '../DateField';
import { getFieldError, getFieldHelperText } from '../../utils';
import { packageConditions } from '../../constants';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, handleChange, setFieldValue, errors, touched } = context;

  return (
    <View style={styles.container} testID="AOSPanelContent">
      <View style={styles.col}>
        <DateField
          label="Received Date"
          value={values.fields.receivedDate}
          onChange={val => setFieldValue('fields.receivedDate', val)}
          style={styles.field}
          hasError={getFieldError('receivedDate', errors, touched)}
          helperText={getFieldHelperText('receivedDate', errors, touched)}
          required={readonly ? false : true}
          touched={touched}
          readonly={readonly}
        />
        {readonly ? <TextInput
          label="Condition of Package"
          value={values.fields.conditionOfPackage ? values.fields.conditionOfPackage.label : ''}
          fullWidth
          readonly={readonly}
        /> : <Select
          label="Condition of Package"
          placeholder={'-None-'}
          options={packageConditions}
          value={values.fields.conditionOfPackage}
          onChange={val => setFieldValue('fields.conditionOfPackage', val)}
          fullWidth
          style={{ width: '100%'}}
          error={getFieldError('conditionOfPackage', errors, touched)}
          helperText={getFieldHelperText('conditionOfPackage', errors, touched)}
          required
        />}

      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyWithBorder : null}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  required: {
    borderWidth: 0
  },
  readonlyWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
