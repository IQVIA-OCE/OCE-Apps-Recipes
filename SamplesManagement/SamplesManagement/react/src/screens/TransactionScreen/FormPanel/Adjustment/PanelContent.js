import React from 'react';
import { View, StyleSheet } from 'react-native';
import { TextInput } from 'apollo-react-native';
import { useFormikContext } from 'formik';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, setFieldValue } = context;

  return (
    <View style={styles.container}>
      <View style={styles.col}>
        <TextInput
          label="Comments"
          onChangeText={text => setFieldValue('fields.comments', text)}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyFieldWithBorder : null}
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      ></View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  readonlyFieldWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
