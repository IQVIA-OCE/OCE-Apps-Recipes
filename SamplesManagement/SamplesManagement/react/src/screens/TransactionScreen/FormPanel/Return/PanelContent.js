import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useFormikContext } from 'formik';
import { TextInput } from 'apollo-react-native';
import DateField from '../DateField';
import { getFieldError, getFieldHelperText } from '../../utils';

const PanelContent = ({ readonly }) => {
  const context = useFormikContext();
  const { values, setFieldValue, errors, touched } = context;

  return (
    <View style={styles.container} testID="ReturnPanelContent">
      <View style={styles.col}>
        <DateField
          label="Shipment Date"
          value={values.fields.shipmentDate}
          onChange={val => setFieldValue('fields.shipmentDate', val)}
          style={styles.field}
          hasError={getFieldError('shipmentDate', errors, touched)}
          helperText={getFieldHelperText('shipmentDate', errors, touched)}
          touched={touched}
          readonly={readonly}
        />
        <TextInput
          label="Tracking Number"
          onChangeText={val => setFieldValue('fields.trackingNumber', val)}
          value={values.fields.trackingNumber}
          readonly={readonly}
          fullWidth
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        <TextInput
          label="Shipment Carrier"
          onChangeText={val => setFieldValue('fields.shipmentCarrier', val)}
          value={values.fields.shipmentCarrier}
          fullWidth
          style={styles.field}
          readonly={readonly}
        />
        <TextInput
          label="Comments"
          onChangeText={val => setFieldValue('fields.comments', val)}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
});

export default PanelContent;
