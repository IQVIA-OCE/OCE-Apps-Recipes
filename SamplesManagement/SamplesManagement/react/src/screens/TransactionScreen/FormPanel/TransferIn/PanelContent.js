import React, { useState, useEffect } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select, Autocomplete } from 'apollo-react-native';
import { useFormikContext } from 'formik';
import DateField from '../DateField';
import { useFetcher, useHandleData } from '../../../../hooks';
import {
  fetchAllUsers,
  fetchUserLocations,
  fetchUserTerritory,
} from '../../../../api/SampleTransaction';
import {
  normalizeUsers,
  normalizeLocations,
  getFieldError,
  getFieldHelperText,
} from '../../utils';
import { packageConditions } from '../../constants';

const PanelContent = ({ readonly, isLoadingDetails }) => {
  const context = useFormikContext();
  const { values, handleChange, setFieldValue, errors, touched } = context;

  const [users] = useFetcher(fetchAllUsers, normalizeUsers);
  const fromSalesRepId = values.fields.fromSalesRep
    ? values.fields.fromSalesRep['value']
    : null;
  const userId =
    values.fields.user && values.fields.user.Id ? values.fields.user.Id : null;

  const [locations, locationsActions] = useFetcher(
    async () => await fetchUserLocations(userId),
    normalizeLocations
  );
  const [territory, territoryActions] = useFetcher(
    async () => await fetchUserTerritory(fromSalesRepId)
  );

  useEffect(() => {
    territoryActions.handleFetch();
  }, [values.fields.fromSalesRep]);

  useEffect(() => {
    if (values.fields.user.Id) {
      locationsActions.handleFetch();
    }
  }, [values.fields.user.Id]);

  useEffect(() => {
    const fromSalesRepTerritory =
      territory.data && territory.data.length
        ? territory.data[0].OCE__Territory__c
        : '';
    setFieldValue('fields.fromSalesRepTerritory.name', fromSalesRepTerritory);
  }, [territory]);

  return (
    <View style={styles.container} testID="TransferInPanelContent">
      <View style={styles.col}>
      {isLoadingDetails ? null : 
        useHandleData(users)(data => {
          return readonly ? (
            <TextInput
              label="From Sales Rep"
              value={
                values.fields.fromSalesRep
                  ? values.fields.fromSalesRep.label
                  : ''
              }
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Autocomplete
              label="From Sales Rep"
              helperText={getFieldHelperText('fromSalesRep', errors, touched)}
              error={getFieldError('fromSalesRep', errors, touched)}
              placeholder=""
              source={data}
              onChange={val => {
                setFieldValue('fields.fromSalesRep', val);
              }}
              singleSelect
              singleSelectValue={values.fields.fromSalesRep}
              fullWidth
              required
              style={styles.field}
            />
          );
        })}
        {useHandleData(locations)(data => {
          return readonly ? (
            <TextInput
              label="Ship To"
              value={values.fields.shipTo ? values.fields.shipTo.label : ''}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Select
              label="Ship To"
              placeholder={'-None-'}
              options={data}
              value={values.fields.shipTo}
              onChange={val => setFieldValue('fields.shipTo', val)}
              fullWidth
              style={styles.field}
              error={getFieldError('shipTo', errors, touched)}
              helperText={getFieldHelperText('shipTo', errors, touched)}
              required
            />
          );
        })}

        <DateField
          label="Received Date"
          value={values.fields.receivedDate}
          onChange={val => setFieldValue('fields.receivedDate', val)}
          style={readonly ? styles.readonlyField : styles.field}
          hasError={getFieldError('receivedDate', errors, touched)}
          helperText={getFieldHelperText('receivedDate', errors, touched)}
          required={readonly ? false : true}
          touched={touched}
          readonly={readonly}
        />

        {values.fields.relatedTransactionName ? (
          <View>
            <TextInput
              label="Shipment Carrier"
              value={values.fields.shipmentCarrier}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          </View>
        ) : null}
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        {useHandleData(territory)(() => {
          return (
            <TextInput
              label="From Sales Rep Territory"
              value={values.fields.fromSalesRepTerritory.name}
              fullWidth
              style={styles.readonlyField}
              readonly
            />
          );
        })}

        {values.fields.relatedTransactionName ? (
          <View>
            <TextInput
              label="Shipment Date"
              value={values.fields.shipmentDate}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
            <TextInput
              label="Tracking Number"
              value={values.fields.trackingNumber}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          </View>
        ) : null}

        {readonly ? (
          <TextInput
            label="Condition of Package"
            style={styles.readonlyField}
            value={
              values.fields.conditionOfPackage
                ? values.fields.conditionOfPackage.label
                : ''
            }
            fullWidth
            readonly
          />
        ) : (
          <Select
            label="Condition of Package"
            placeholder={'-None-'}
            options={packageConditions}
            value={values.fields.conditionOfPackage}
            onChange={val => setFieldValue('fields.conditionOfPackage', val)}
            fullWidth
            style={styles.field}
            error={getFieldError('conditionOfPackage', errors, touched)}
            helperText={getFieldHelperText(
              'conditionOfPackage',
              errors,
              touched
            )}
            required
          />
        )}

        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          numberOfLines={5}
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyWithBorder : null}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  readonlyField: {
    marginBottom: 18,
  },
  readonlyWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
