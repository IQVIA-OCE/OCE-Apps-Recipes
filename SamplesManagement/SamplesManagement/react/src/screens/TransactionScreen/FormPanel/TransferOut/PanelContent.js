import React, { useEffect } from 'react';
import { View, StyleSheet, Text } from 'react-native';
import { TextInput, Select, Autocomplete } from 'apollo-react-native';
import { useFormikContext } from 'formik';
import DateField from '../DateField';
import { useFetcher, useHandleData } from '../../../../hooks';
import {
  fetchAllUsers,
  fetchUserLocations,
  fetchUserTerritory,
} from '../../../../api/SampleTransaction';
import {
  normalizeUsers,
  normalizeLocations,
  getFieldError,
  getFieldHelperText,
} from '../../utils';

const PanelContent = ({ readonly }) => {
  const {
    values,
    handleChange,
    setFieldValue,
    errors,
    touched,
  } = useFormikContext();

  const [users] = useFetcher(fetchAllUsers, normalizeUsers);

  const toSalesRepId = values.fields.toSalesRep
    ? values.fields.toSalesRep['value']
    : null;

  const [locations, locationsActions] = useFetcher(
    async () => await fetchUserLocations(toSalesRepId),
    normalizeLocations
  );
  const [territory, territoryActions] = useFetcher(
    async () => await fetchUserTerritory(toSalesRepId)
  );

  useEffect(() => {
    locationsActions.handleFetch();
    territoryActions.handleFetch();
  }, [values.fields.toSalesRep]);

  useEffect(() => {
    const toSalesRepTerritory =
      territory.data && territory.data.length
        ? territory.data[0].OCE__Territory__c
        : '';
    setFieldValue('fields.toSalesRepTerritory.name', toSalesRepTerritory);
  }, [territory]);

  return (
    <View style={styles.container}>
      <View style={styles.col}>
        {useHandleData(users)(data => {
          return readonly ? (
            <TextInput
              label="To Sales Rep"
              value={
                values.fields.toSalesRep ? values.fields.toSalesRep.label : ''
              }
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Autocomplete
              label="To Sales Rep"
              helperText={getFieldHelperText('toSalesRep', errors, touched)}
              error={getFieldError('toSalesRep', errors, touched)}
              placeholder=""
              source={data}
              onChange={val => {
                setFieldValue('fields.shipTo', null);
                setFieldValue('fields.toSalesRep', val);
              }}
              singleSelect
              singleSelectValue={values.fields.toSalesRep}
              fullWidth
              required
              style={styles.field}
            />
          );
        })}
        {useHandleData(locations)(data => {
          return readonly ? (
            <TextInput
              label="Ship To"
              value={values.fields.shipTo ? values.fields.shipTo.label : ''}
              fullWidth
              readonly
              style={styles.readonlyField}
            />
          ) : (
            <Select
              label="Ship To"
              placeholder={'-None-'}
              options={data}
              value={values.fields.shipTo}
              onChange={val => setFieldValue('fields.shipTo', val)}
              fullWidth
              style={{ width: '100%', marginBottom: 15 }}
              error={getFieldError('shipTo', errors, touched)}
              helperText={getFieldHelperText('shipTo', errors, touched)}
              disabled={values.fields.toSalesRep == null || users.loading}
              required
            />
          );
        })}

        <DateField
          label="Shipment Date"
          value={values.fields.shipmentDate}
          onChange={val => setFieldValue('fields.shipmentDate', val)}
          style={styles.field}
          hasError={getFieldError('shipmentDate', errors, touched)}
          helperText={getFieldHelperText('shipmentDate', errors, touched)}
          touched={touched}
          readonly={readonly}
          required
        />

        <TextInput
          label="Shipment Carrier"
          onChangeText={handleChange('fields.shipmentCarrier')}
          value={values.fields.shipmentCarrier}
          fullWidth
          error={getFieldError('shipmentCarrier', errors, touched)}
          helperText={getFieldHelperText('shipmentCarrier', errors, touched)}
          required
          readonly={readonly}
          style={readonly ? styles.readonlyField : styles.field}
        />
      </View>
      <View
        style={{
          flex: 1,
        }}
      >
        {useHandleData(territory)(() => {
          return (
            <TextInput
              label="To Sales Rep Territory"
              value={values.fields.toSalesRepTerritory.name}
              fullWidth
              style={[styles.field, { borderWidth: 0, opacity: 1 }]}
              readonly
              editable={false}
            />
          );
        })}

        <TextInput
          label="Tracking Number"
          onChangeText={handleChange('fields.trackingNumber')}
          value={values.fields.trackingNumber}
          error={getFieldError('trackingNumber', errors, touched)}
          helperText={getFieldHelperText('trackingNumber', errors, touched)}
          fullWidth
          required
          readonly={readonly}
          style={readonly ? styles.readonlyField : styles.field}
        />
        <TextInput
          label="Comments"
          onChangeText={handleChange('fields.comments')}
          value={values.fields.comments}
          multiline
          fullWidth
          readonly={readonly}
          style={readonly ? styles.readonlyWithBorder : styles.field}
        />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 15,
    paddingHorizontal: 30,
    marginBottom: 10,
  },
  col: {
    borderRightWidth: 1,
    borderRightColor: 'rgb(221, 219, 218)',
    flex: 1,
    marginRight: 20,
    paddingRight: 20,
  },
  field: {
    marginBottom: 15,
  },
  readonlyField: {
    marginBottom: 18,
  },
  readonlyWithBorder: {
    marginBottom: 10,
    borderBottomWidth: 0.5,
    borderColor: '#D9D9D9',
  },
});

export default PanelContent;
