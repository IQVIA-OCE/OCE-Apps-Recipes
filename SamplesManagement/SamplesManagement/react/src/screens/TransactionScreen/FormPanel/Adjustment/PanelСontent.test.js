import React from 'react';
import PanelContent from './PanelContent';
import { Formik } from 'formik';
import { TextInput } from '@oce-apps/apollo-react-native';
import { render, act } from '@testing-library/react-native';

describe('PanelContent', () => {
  it('should render properly', () => {
    const { getByText } = render(
      <Formik
        initialValues={{
          fields: {
            comments: '',
          },
        }}
      >
        <PanelContent />
      </Formik>
    )

    expect(getByText(/Comments/)).toBeTruthy();
  });

  it('should call Input onchange', async () => {
    const { UNSAFE_root } = render(
      <Formik
        initialValues={{
          fields: {
            comments: '',
          },
        }}
      >
        <PanelContent />
      </Formik>
    )

    act(() => {
      UNSAFE_root.findByType(TextInput).props.onChangeText('test');
    });

    expect(UNSAFE_root.findByType(TextInput).props.value).toEqual('test')
  });
});
