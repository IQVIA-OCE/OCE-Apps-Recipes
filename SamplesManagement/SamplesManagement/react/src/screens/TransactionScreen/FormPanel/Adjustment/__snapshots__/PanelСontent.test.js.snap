// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelContent should call Input onchange 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 15,
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          null,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Comments
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "minHeight": 80,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 80,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value="test"
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`PanelContent should render properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 10,
      "padding": 15,
      "paddingHorizontal": 30,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 20,
        "paddingRight": 20,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "width": "100%",
          },
          null,
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Comments
      </Text>
      <View
        style={
          Array [
            Object {
              "borderColor": "#D9D9D9",
              "borderRadius": 4,
              "borderWidth": 1,
            },
            Object {},
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 80,
              "minHeight": 80,
              "overflow": "hidden",
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            maxResizableWidth={750}
            minHeight={60}
            minWidth={0}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            paddingLeft={8}
            paddingRight={8}
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#0768FD"
            style={
              Array [
                Object {
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 5,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": undefined,
                  "height": 80,
                  "minHeight": 0,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingVertical": 5,
                  "textAlignVertical": "center",
                  "width": "100%",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
